Commands/MessageCount/MessageCountSystem.go

```diff
@@ -5,13 +5,37 @@
 		-msgcount
 */}}
 
-{{$x:=dbIncr .User.ID "msgCount" 1}}{{$list:=""}}
-{{if reFind `\A-m(?:essages?|sg)?c(?:ount)?` .Message.Content}}
-	 You have typed {{$x}} messages.
-{{end}}
-{{if reFind `\A-t(?:op)?m(?:essages?|sg)?` .Message.Content}}
-	{{range dbTopEntries "msgCount" 10 0}}
-		{{- $list =print $list .User.Username " - " .Value "\n" -}}
-	{{end}}
-	{{sendMessage nil (cembed "title" "Message Count Leaderboard" "description" $list)}}
-{{end}}
+{{/* DONT TOUCH UNLESS YOU KNOW WHAT UR DOING c: */}}
+{{/* define variables */}}
+{{ $avatar := (joinStr "" "https://cdn.discordapp.com/avatars/" (toString .User.ID) "/" .User.Avatar ".png") }}
+{{ $x: = dbIncr .User.ID "msgCount" 1 }}{{ $list:="" }}
+{{ $embed := cembed "thumbnail" (sdict "url" $avatar) "title" "Message Counter" "description" (print .User.Mention " have sent " $x  " message(s) ") "color" 0xFF0000 }}
+{{ if reFind `\A\-m(?:essages?|sg)?c(?:ount)?` .Message.Content }} {{/* msgcount command */}}
+	{{ sendMessage nil $embed }}
+{{ end }}	
+{{ if reFind `\A\-t(?:op)?m(?:essages?|sg)?` .Message.Content }} {{/* topmessages command */}}
+{{/* better leaderboard, made by joe, modified by blox c: */}}
+{{ $page := 1 }} {{/* Default page to start at */}}
+{{ $skip := mult (sub $page 1) 10 }} {{/* Amount of entries to skip */}}
+{{ $users := dbTopEntries "msgCount" 10 $skip }} {{/* Retrieve the relevant DB entries with the parameters provided */}}
+{{ if not (len $users) }}
+	There were no users on that page! {{/* If there were no users, return */}}
+{{ else }}
+	{{ $rank := $skip }} {{/* Instantiate rank variable with value of $skip */}}
+	{{ $display := "" }} {{/* The description for the leaderboard description */}}
+	{{- range $users -}}
+		{{ $coins := toInt .Value }} {{/* Messages for this user entry */}}
```

Comment:
use - to remove whitespace within a range, follow this format:
{{range ...}}
   {{- ...}}
   {{- ...}
   {{- ... -}}
{{end}}